docs/source/doc-conda-requirements.yml

```diff
@@ -1,9 +1,10 @@
-name: httomo
+name: httomo-docs
 channels:
   - conda-forge
   - defaults
 dependencies:
-  - python=3.9
+  - python
```

Comment:
After updating my old conda env with the new file and trying to build the docs locally, I get a ton of warnings like the following during the build:
```
WARNING: autodoc: failed to import module 'cli' from module 'httomo'; the following exception was raised:
Traceback (most recent call last):
  File "/dls/science/users/twi18192/conda-envs/docs-httomo-py310/lib/python3.12/site-packages/sphinx/ext/autodoc/importer.py", line 62, in import_module
    return importlib.import_module(modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/dls/science/users/twi18192/conda-envs/docs-httomo-py310/lib/python3.12/importlib/__init__.py", line 90, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 995, in exec_module
  File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
  File "/dls/science/users/twi18192/httomo/httomo/cli.py", line 16, in <module>
    from httomo.transform_layer import TransformLayer
  File "/dls/science/users/twi18192/httomo/httomo/transform_layer.py", line 4, in <module>
    from httomo.method_wrappers import make_method_wrapper
  File "/dls/science/users/twi18192/httomo/httomo/method_wrappers/__init__.py", line 1, in <module>
    from httomo.runner.method_wrapper import MethodWrapper
  File "/dls/science/users/twi18192/httomo/httomo/runner/method_wrapper.py", line 2, in <module>
    from httomo.runner.dataset import DataSetBlock
  File "/dls/science/users/twi18192/httomo/httomo/runner/dataset.py", line 2, in <module>
    from typing_extensions import TypeAlias
  File "/dls/science/users/twi18192/conda-envs/docs-httomo-py310/lib/python3.12/site-packages/typing_extensions.py", line 1174, in <module>
    class TypeVar(typing.TypeVar, _DefaultMixin, _root=True):
TypeError: type 'typing.TypeVar' is not an acceptable base type
```

I think this is because the python version in the env got updated to 3.12 when I ran `mamba env update --file ...` due to the removal of `=3.9`, and the httomo code only supports python 3.9 and 3.10, so a python 3.12 interpreter parses the code incorrectly.

Would it be ok to put a version constraint back on `python` to avoid such warnings, the docs build output gets flooded with this stuff and makes it hard to see other potential issues during the build?

Reply:
interesting, it was certainly OK for me even on python 3.12. Don't know why the docstrings compilation should be linked to the actual httomo Python version? But if doesn't matter a lot, so fine with 3.9 constraint back.

Reply:
That's interesting that you didn't get these warnings with python 3.12, maybe something additional is going on locally for my particular case.

The python version matters because sphinx actually executes the imports in the source code when it's building the docs from the source code (hence, why there's some packages being mocked, to avoid import errors): https://github.com/DiamondLightSource/httomo/blob/98e9003ecf0b31fe3babf50a1b8f848a8bb50326/docs/source/conf.py#L19-L32

and so when a python 3.12 interpreter is used to do the imports in the httomo code, the above errors would occur I believe.

Sphinx catches the errors from the imports and produces its own warning, that python failed to do the import.

Reply:
(It's only a warning, it's not an error, so the docs build ok. But it's just that there's a lot of the warnings when I did the build locally, and it was hard to read the output and determine if there were any other issues, because there are plenty of other warnings in the docs build - at least, for me locally)